src/assemblyItemsBrowser.js

```diff
@@ -192,13 +198,58 @@ module.exports = React.createClass({
 		}
 		
 		var currentAddress = this.state.currentAddress
-		if (props.vmTrace[vmTraceIndex].address) // there's a new context
-		{
-			var stack = this.state.callStack[vmTraceIndex]
-			currentAddress = stack[stack.length - 1]
-			this.resolveAddress(currentAddress, props)
-			Object.assign(stateChanges, { "currentAddress": currentAddress })
-		}
+        var addressIndex = this.shouldUpdateStateProperty("address", vmTraceIndex, previousIndex, props.vmTrace)
+        if (addressIndex > -1)
+        {
+            if (this.state.currentSelected === -1) // beginning of the trace
+            {
+                currentAddress = props.vmTrace[addressIndex].address
+                if (props.transaction.to === null) // contract creation
+                {
+                    this.cacheCode(currentAddress, props.transaction.input)
+                    Object.assign(stateChanges,
+                    {
+                        "currentAddress": currentAddress,
+                        "selectedInst": this.state.instructionsIndexByBytesOffset[currentAddress][props.vmTrace[vmTraceIndex].pc],
+                        "currentSelected": vmTraceIndex
+                    })
+                }
+                else
+                {
+                    var self = this
+                    this.loadCode(currentAddress, props, function(code) // contract call
+                    {
+                        self.setState({
```

Comment:
But that will generate problems for e.g. the slider.
I think it would be better to associate the callback result with the ingoing parameters and throw away the result if we are already farther.